flax/errors.py

```diff
@@ -117,4 +116,44 @@ def __call__(self, x):
         return x
   """
   def __init__(self, scope_name):
-    super().__init__(f'Duplicate use of scope name: "{scope_name}"')
\ No newline at end of file
+    super().__init__(f'Duplicate use of scope name: "{scope_name}"')
+
+
+class JaxOmnistagingError(FlaxError):
+  """
+  The Flax linen API requires JAX omnistaging to be enabled. In order to enable
+  this, add this to your imports::
+    
+    from jax.config import config
+    config.enable_omnistaging()
+  """
+  def __init(self):
+    super().__init__(f'Flax Linen requires Omnistaging to be enabled')
+
+
+class UnboundCompactModuleError(FlaxError):
+  """
+  This error occurs when you are trying to call a Module directly, rather than
+  through :meth:`Module.apply() <flax.linen.Module.apply>`. For instance, the 
+  error will be raised when trying to run this code::
+
+    from flax import linen as nn
+    import jax.numpy as jnp
+
+    test_dense = nn.Dense(10)
+    test_dense(jnp.ones((5,5)))
+
+  Instead, you should pass the variables (parameters and other state) via 
+  :meth:`Module.apply() <flax.linen.Module.apply>` (or use 
+  :meth:`Module.init() <flax.linen.Module.init>` to get initial variables)::
+
+    # Create the initialized variables with a random key for inits:
+    params = test_dense.init(random.PRNGKey(0), jnp.ones((5,5)))['params']
```

Comment:
I think this code doesn't work. It should be `variables = ` and don't do `['params']` at the end

Reply:
Ah, I get a different error then -- `setup` doesn't run so the submodule I defined isn't there

https://colab.research.google.com/drive/1SsRzkBoGqf1DNYRVseh5QxjA_SwCb7gb

Reply:
I don't seem to get a different error when I do as you suggested:

```
from flax import linen as nn
import jax.numpy as jnp

test_dense = nn.Dense(10)

 # Create the initialized variables with a random key for inits:
import jax.random
vars = test_dense.init(random.PRNGKey(0), jnp.ones((5,5)))

# Apply the NN to the variables + input to get output.
y = test_dense.apply(vars, jnp.ones((5,5)))
```